packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,287 @@
+  },
+  {
+    section: 'Previous',
+    timeframeText: 'Week',
+    timeframeLength: 60 * 60 * 24 * 90
+  },
+  {
+    section: 'Previous',
+    timeframeText: 'Month',
+    timeframeLength: 60 * 60 * 24 * 90
+  },
+  {
+    section: 'Previous',
+    timeframeText: 'Quarter',
+    timeframeLength: 60 * 60 * 24 * 90
+  }
+]
+
+export default {
+  name: 'KTimePicker',
+  components: {
+    KButton,
+    KPop,
+    KSegmentedControl,
+    DatePicker: () => import('v-calendar/lib/components/date-picker.umd')
+  },
+  props: {
+    value: {
+      type: Object,
+      required: true
+    }
+  },
+
+  data () {
+    return {
+      allowedTimePeriods,
+      mode: this.mode || 'custom',
+      selectimedTimeRange: this.selectimedTimeRange,
+      selectedTimeframe: this.value || this.allowedTimePeriods[0],
+      modelConfig: {
+        type: 'number'
+      }
+    }
+  },
+
+  computed: {
+    showCalendar () {
+      return (this.mode === 'custom')
+    }
+  },
+
+  watch: {
+    selectimedTimeRange: {
+      handler (newValue, oldValue) {
+        // console.warn('>>>> selectimedTimeRange watcher <<<')
+        // console.warn(newValue)
+      },
+      immediate: true
+    }
+  },
+
+  methods: {
+    /**
+     * TODO: Given a relative time frame, determine `start` and `end` timestamps in UTC
+     * https://bloop.ai/search/how-to-use-date-fns-library-round-to-nearest-minutes-in-javascript
+     * @param {*} timeframe
+     */
+    changeTimeframe (timeframe) {
+      this.selectedTimeframe = timeframe
+    },
+    clearSelection () {
+      this.selectimedTimeRange = ''
+      this.selectedTimeframe = this.allowedTimePeriods[0]
+    },
+    submitTimeFrame () {
+      console.warn('>>> submitTimeFrame')
+      let start = 0
+      let end = 0
+
+      // If calendar currently shown, send start/end values
+      if (this.showCalendar) {
+        start = Math.floor(this.selectimedTimeRange.start / 1000)
+        end = Math.floor(this.selectimedTimeRange.end / 1000)
+      } else {
+        end = getUnixTime(roundToNearestMinutes(Date.now()))
+        start = end - this.selectedTimeframe.timeframeLength
+      }
+
+      console.log({ start, end })
+
+      this.$emit('changed', { start, end })
+    }
+  }
+}
+
+</script>
+
+<style lang="scss">
+@import '~@kongponents/styles/variables';
+
+$margin: .2rem;
+
+.analytics-timepicker {
+  // TODO: consider moving to a new button theme
+  .timepicker-input {
+    padding: var(--spacing-sm) var(--spacing-xl) var(--spacing-sm) 40px !important;
+    background-image: url("data:image/svg+xml,%3Csvg width='16' height='16' viewBox='0 0 16 16' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill-rule='evenodd' clip-rule='evenodd' d='M14 14H2V2H4V4H6V2H10V4H12V2H14V14ZM14 0H2C0.895 0 0 0.896 0 2V14C0 15.104 0.895 16 2 16H14C15.104 16 16 15.104 16 14V2C16 0.896 15.104 0 14 0ZM7 12H9V10H7V12ZM10 9H12V7H10V9ZM7 9H9V7H7V9ZM4 9H6V7H4V9ZM4 12H6V10H4V12Z' fill='%236F7787'/%3E%3C/svg%3E%0A");
```

Comment:
Do we need this custom calendar icon? My preference would be to add a new calendar icon to `KIcon` and then use that, if possible.

Reply:
It's not finalized yet :/ I'll check and get back to you on that.